### ISSUE TITLE
ERR_SSL_PROTOCOL_ERROR

### ISSUE THREAD

**Issue by Zitzabis** [minotar/imgd#179]
Due to recent versions of Chrome, your site is now returning a ERR_SSL_PROTOCOL_ERROR code for any resource loaded from your site except for the home page.

Planet Minecraft Chat pulls it's player heads using your service and they are currently returning errors in the console.

[http://i.imgur.com/0FrdBsz.png](http://i.imgur.com/0FrdBsz.png)

**Comment by LukeHandle** [comment]
Hey, thanks for reaching out.

Just to confirm, which Chrome version? And, if you try and open both the following the links, do you also see errors?

https://minotar.net/

https://minotar.net/avatar/LukeHandle.png

All traffic is served via Cloudflare (and SSL is re-encrypted)  so I'm very intrigued as to the errors you are seeing. Are you accessing the site from a restricted network - one which might be performing some sort of SSL interception (eg. Corporate, School/Uni)? 

I would anticipate a few things:
* Cloudflare strange issue
* Intermittent issue on our SSL proxy (Nginx)
* Your traffic is passing through another party which is mangling the SSL traffic (and a configuration in our Cloudflare doesn't help)

**Comment by Zitzabis** [comment]
I am running version 56.0.2924.87 of Chrome.

The main page works using HTTP, but not HTTPS.

The second link works the same way.

I am accessing it from a restricted network however I spoke to the network admin today about it. We have the potential to alter the SSL traffic but we have no filters being applied to them so there should be no interference. It works just fine on mobile though within the restricted network so I don't believe it lies on our own.

From what I understand (and I found very little documentation on this matter), Chrome devs recently changed some of their settings as to what they do and do not accept in terms of security certificates. Essentially, they no longer accept SHA1 as it has potential vulnerabilities. However, SHA2 does work.
Just yesterday the admin finally purchased new certificates for all of our sites and upgraded them to SHA2.

The confusing part (for both of us no doubt) lies with the fact that you do not appear to be using SHA1. Instead it's SHA256 from what I can see.

Unfortunately that's all the info I can give.

**Comment by clone1018** [comment]
I ran a couple of Qualys tests and it didn't find any potential issues: https://www.ssllabs.com/ssltest/analyze.html?d=minotar.net

**Comment by Zitzabis** [comment]
I agree clone, can't see any errors either using external services.

The best article I can find is here:
[https://security.googleblog.com/2016/11/sha-1-certificates-in-chrome.html](https://security.googleblog.com/2016/11/sha-1-certificates-in-chrome.html)

**Comment by jomo** [comment]
@Zitzabis most likely a problem with traffic interception in your network. I assume you're getting this error on _any_ site behind Cloudflare. Try https://blog.filippo.io/

**Comment by Zitzabis** [comment]
@jomo 
I strongly doubt it's my network. I can access all pages normally using Firefox and IE. Both of which would be passing traffic through the same system.

That blog returns the same error in Chrome.
Works fine in Firefox and IE.

[http://i.imgur.com/shLPsQR.png](http://i.imgur.com/shLPsQR.png)
Seen in the screenshot is FF in the background, Chrome in the foreground and IE in the middle.

Cloudflare being the root of the problem is possible. However, I am a moderator for [http://www.planetminecraft.com/](http://www.planetminecraft.com/) and I have yet to see any issues with our site despite the fact that we also use Cloudflare. The only issue occurs when sourcing images from your system.

Could a change in how we call your API have any impact? HTTP vs. HTTPS

**Comment by jomo** [comment]
SSL is what HTTPS uses, so you'll only get that error when loading a page via HTTPS. https://planetminecraft.com is probably going to fail as well. It's not a problem with Cloudflare, Minotar, nor Chrome itself. I'll help you debugging though :)

My guess is you have a proxy, a plugin, (or some other tool, like anti virus) intercepting traffic (for Chrome only) and it can't handle the ciphers Cloudflare uses or the likes. It's also possible you have some weird Chrome flags (`chrome://flags`) on something non-default.

Can you post the "Given Cipher Suites" of https://www.howsmyssl.com/ in FF and Chrome and if any of those 6 categories are not "Good"?

**Comment by Zitzabis** [comment]
[https://planetminecraft.com/](https://planetminecraft.com/) works just fine.

Chrome runs Good across the board.
IE returns Improvable with Session Ticket Support.
Firefox runs Good across the board.

I have looked at Chrome flags and all of those are set to default. I have done so multiple times.

These are all of the Chrome Cipher Suites.

- TLS_GREASE_IS_THE_WORD_0A
- TLS_AES_128_GCM_SHA256
- TLS_AES_256_GCM_SHA384
- TLS_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
- TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
- TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
- TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
- TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
- TLS_RSA_WITH_AES_128_GCM_SHA256
- TLS_RSA_WITH_AES_256_GCM_SHA384
- TLS_RSA_WITH_AES_128_CBC_SHA
- TLS_RSA_WITH_AES_256_CBC_SHA
- TLS_RSA_WITH_3DES_EDE_CBC_SHA

I work in the IT department for my university. We have been receiving calls from various people involving the same issue I see but using our sites. Hence why the network admin has made an upgrade on the certificates. These are all people external to the system and all roots to the problem is Chrome. The suggested fix we have been giving them is to use FF or IE.

**Comment by LukeHandle** [comment]
Interesting, though not really a coincidence, but I actually had to troubleshoot a SSL MITM (Man in the Middle) issue the other day.

In that case though, it was a corporate provided computer which had an internally/company signed root certificate installed. The MITM device they locally used then detected an SSL connection to a site (servername via ClientHello), generated a certificate for it on the fly using the internal CA and presented it to the user. This then allowed them to see the SSL traffic and as their MITM device had the certificate installed.

The problem for the user (and why it was highlighted as an issue) was that the internal CA was signed with a SHA1 causing Chrome to complain.

Now based on "we have the potential to alter the SSL traffic", it sounds like this sort of MITM device is in-place on your network? The next important question is whether the Chrome computer you are using was supplied by your Uni? Did they ask you to add any specific certificate to your local trust store when you first set it up?

The next test I would be interested in you performing would be verifying the certificate/cipher details that you get on Firefox when browsing the site.

Clicking the Green padlock, going to "More Information" (alternatively, right-click the page, "Page Info" -> "Security" tab) and you are presented with a some brief information about the certificate and the connection your client has made to the site.

Are you able to screenshot that window (I'm interested in the ciphers/protocol that Firefox connected with - are these expected/secure). Further to this, clicking the "View Certificate" button will show more information about the certificate sent to you by the webserver. Are you able to screenshot this window (which includes the SHA-256 Fingerprint) so we can verify that you are being presented with our/Cloudflare SSL and not another.

I would expect the SHA-256 Fingerprint to be:
`13:59:2D:C9:89:93:3A:F0:79:DD:E6:84:8D:96:08:DD:0F:F8:90:26:E1:81:D0:CE:18:7F:0A:82:6F:13:D5:1F`

**Comment by Zitzabis** [comment]
I managed to resolve the issue (and I'm really sorry about this drawn out process).

My university requires that we install a small program called SafeConnect that runs transparently in the background. It helps ensure that the anti-virus is up to date and automatic Windows updates are turned on. However, on top of that it also monitors what programs you are running (this is to prevent evasion via Tor or by using P2P software for downloads). This is all to endure the ecosystem is running normally.
I have for years now classified this small piece of software as a virus due to it's nature of operations. I have altered Chrome's user agent so as to appear to be running behind a Ubuntu system so as to bypass the whole mandatory program downloading option as I am strongly against it. I recently had to undo that though as the IT department voiced their displeasure on the matter.

While the issue I had did not lie with your system in the slightest, it is peculiar that your system was targeted by it. One issue solved, another raised. Time to go speak to the head of the IT department and complain profusely about what they are doing.

Thank you very much @LukeHandle! You gave me that final piece of the puzzle to look into that area.
